tests/acceptance/settings-page/billing-test.js

```diff
@@ -0,0 +1,188 @@
+import { module, test } from 'qunit';
+import { setupApplicationTest } from 'codecrafters-frontend/tests/helpers';
+import { signIn, signInAsSubscriber, signInAsVipUser } from 'codecrafters-frontend/tests/support/authentication-helpers';
+import testScenario from 'codecrafters-frontend/mirage/scenarios/test';
+import billingPage from 'codecrafters-frontend/tests/pages/settings/billing-page';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { setupWindowMock } from 'ember-window-mock/test-support';
+import percySnapshot from '@percy/ember';
+import { format } from 'date-fns';
+
+module('Acceptance | settings-page | billing-test', function (hooks) {
+  setupApplicationTest(hooks);
+  setupMirage(hooks);
+  setupWindowMock(hooks);
+
+  test('membership section shows correct plan for subscriber with active subscription', async function (assert) {
+    testScenario(this.server);
+    signInAsSubscriber(this.owner, this.server);
+    const subscription = this.server.schema.subscriptions.first();
+    subscription.update('pricingPlanName', 'Yearly Plan');
+
+    await billingPage.visit();
+
+    assert.ok(billingPage.membershipSection.isVisible, 'membership section is visible');
+    assert.ok(billingPage.membershipSection.text.includes('Membership active'), 'shows active plan');
+
+    assert.ok(
+      billingPage.membershipSection.text.includes(format(new Date(new Date().getTime() + 24 * 60 * 60 * 1000), 'PPPp')),
```

Comment:
@Arpan-206 try to avoid logic (like using `format`) in tests, it's easy for mistakes to creep in and tests just "pass" because the use the same flawed implementation. Related: https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html

As an example, if we had to switch from `date-fns` to a different library this test wouldn't be useful since we'd have to change the test too and won't be able to rely on for the refactor. 

The more robust way to test this would be: `text.includes('Your membership is valid until 23 January 2023 12:45PM')` (i.e. test the full sentence). You can set a `expiresAt` date on the subscription ahead of time so that the string is fixed